code/__HELPERS/unsorted.dm

```diff
@@ -683,15 +683,34 @@ Turf and target are seperate in case you want to teleport some distance from a t
 	var/target_loc = target.loc
 	var/holding = user.get_active_hand()
 	var/timefraction = round(time/numticks)
-	for(var/i = 0, i<numticks, i++)
+	var/image/progbar
+	for(var/i = 1 to numticks)

```

Comment:
Put this part inside another internal helper proc, then you can do the progbar bits around the outside of the method